### ISSUE TITLE
Keyword 'UTYPE' not found error, but I don't have UNCERT in the FITS header

### ISSUE THREAD

**Issue by mjibrown** [astropy/ccdproc#664]
I've been coming up with an error and cannot figure out what's going on.

I've been reducing some images in astropy on my Mac laptop, using astropy 3.0.2 and ccdproc 1.3.0. However, when I try to open the resulting FITS images on another computer using astropy 3.1.2 and ccdproc 1.3.0.post1 I get the error message KeyError: "Keyword 'UTYPE' not found."

I've looked for the UNCERT and UTYPE keywords in FITS images created by the two versions of astropy+ccdproc but have found nothing. What am I missing? 

A screen cap of the error messages is attached. 

Some of the images that generate the error message can be downloaded via https://drive.google.com/open?id=1kMdUzswfR7uxMKOD3PYYn_IJVwwZ0fYu

![screen_shot_2019-03-18_at_9 01 09_am](https://user-images.githubusercontent.com/44258048/54724307-b1c72a80-4bbe-11e9-8399-5ffde0091762.png)

**Comment by mwcraig** [comment]
@mjibrown -- thanks for the report and the link to the images. I'll take a look in the next column of days.

**Comment by mjibrown** [comment]
This may or may not be related. Yesterday I used Astrometry.Net to attach a WCS to an image from the same telescope and the astropy 3.0.2 and ccdproc 1.3.0 has trouble reading all the WCS keywords (ds9 seemed fine).

The images both with and without the WCS are at https://drive.google.com/open?id=1Xhe3zNzjYvdMIQOJpQYZ0v5Zp1KqNn0r 

I ran this past Dustin Lang, who noticed the headers have some blank lines in them, which shouldn't be a problem but seemed unusual.

![Untitled 11-1](https://user-images.githubusercontent.com/44258048/54856909-cfc49480-4d50-11e9-8f22-62b1fb37647d.png)

**Comment by mwcraig** [comment]
@mjibrown -- good news and bad news.

The bad news is that a bug introduced in astropy 3.1 is what is preventing you from opening the files you wrote with astropy 3.0.2. I'm about to open a pull request to fix that, but it will be little while until there is a new release.

The good new is that there is a workaround. If you replace

```python
scim = [CCDData.read(fn) for fn in ifc.files_filtered(PICTTYPE=1)]
```

with 

```python
scim = [CCDData.read(fn, key_uncertainty_type='naxis') for fn in ifc.files_filtered(PICTTYPE=1)]
```

then the files should open, so there is a temporary workaround.

In astropy 3.1 a couple of new uncertainty types were introduced and to read code was modified to check which uncertainty type had been written. Though the intent was to read files written with older versions, the code checks for a `UTYPE` keyword and doesn't find one in files written with pre-3.1 astropy.

Checking on the WCS issue; I think I know what is going on but want to confirm...

**Comment by mwcraig** [comment]
The issue with the WCS is much easier to fix (though it also uncovers an issue I should address in astropy at some point).

When you read in an image with `CCDData` that has WCS in it a wcs object is automatically created for you and the WCS header keywords are supposed to be completely removed from the header. 

In the sample code you provided, if you replace

```python
w  = WCS(scim.header)
```

with

```python
w = scim.wcs
```

then the remaining code should work.

The issue is that some of the WCS keywords are not being deleted, and the blank lines are a bit odd...but neither of those should affect you in any way if you make the change above.

When the file is written all of the WCS keywords are put back in. The reason we convert to a WCS object and remove the header keywords is that we didn't want to keep both around because modifying one without modifying the other can lead to an inconsistent WCS when the CCDData object is written out.

**Comment by mwcraig** [comment]
I'm closing this; both issues will be fixed in astropy 3.2. If you think it should be reopened please feel free to reopen it, @mjibrown, and thanks again for the detailed report!
